core/context.go

```diff
@@ -0,0 +1,260 @@
+// Copyright © 2020 Solus Project
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package core
+
+// #include <pwd.h>
+// #include <grp.h>
+// #include <unistd.h>
+import "C"
+
+import (
+	"fmt"
+	log "github.com/DataDrake/waterlog"
+	"os/exec"
+	gouser "os/user"
+	"strconv"
+)
+
+const minimumUID = 1000
+const wheelGroup = "sudo"
+
+// Context contains contextual system data, such as groups, users, and active shells, along with paths to certain binaries
+type Context struct {
+	usermod  string
+	groupadd string
+	groupmod string
+	users    []User
+	groups   []gouser.Group
+	shells   []string
+}
+
+// User is a convenience struct with information on a user entry in /etc/passwd
+type User struct {
+	Name     string
+	Groups   []string
+	IsActive bool
+	IsRoot   bool
+	IsAdmin  bool
+}
+
+// NewContext creates an initialized instance of a Context object
+func NewContext() (*Context, error) {
+	return (&Context{}).init()
```

Comment:
Not sure why `init()` needs to be a separate function. You aren't using it anywhere else.